compiler/resolution/functionResolution.cpp

```diff
@@ -950,6 +950,23 @@ bool canInstantiate(Type* actualType, Type* formalType) {
         if (instantiatedFieldsMatch(actualType, formalType)) {
           return true;
         }
+
+        // We may have a 'DecoratedClassType' as one of the arguments
+        // (e.g. anymanaged x1) and a regular class (e.g. shared(x2)) as another.
+        // This would've failed the check above, since anymanaged x1 is not
+        // an aggregate typem and x1 is not an instantiation of shared(x2), either.
+        //
+        // By the above condition, x1 != x2, but instantiation is still possible
+        // if one of the xs instantiates another. Strip the 'owned' etc.
+        // In that case, compare the underlying types.
+        bool isActualAnyManaged = isDecoratedClassType(actualType) &&
+          (actualDec & ClassTypeDecorator::MANAGEMENT_MASK) == ClassTypeDecorator::GENERIC;
+        bool isFormalAnyManaged = isDecoratedClassType(formalType) &&
+          (formalDec & ClassTypeDecorator::MANAGEMENT_MASK) == ClassTypeDecorator::GENERIC;
+        if ((isActualAnyManaged || isFormalAnyManaged) &&
```

Comment:
I would think that this should only consider `isFormalAnyManaged && !isActualAnyManaged`. If the actual is any-managed e.g. `C`, we can't pass it to something with specific manegement e.g. `shared C`

Reply:
What about:

```
proc test(type myTest: genClass(?)) {}
test(genClass(int));
```

(note: the code passes on main, since some other mechanisms must take care of the instantiation). I'd want `anymanaged genClass(int)` to work as an actual to `anymanaged genClass(?)`. At least, to match the existing generic-class-as-type-actual behavior.

Reply:
Seems like maybe the condition should be `isFormalAnyManaged` only.

Reply:
And, I believe, `canInstantiateDecorators` would've already ruled out "any managed actual to shared formal".